app/jobs/in_person/enrollments_ready_for_status_check_job.rb

```diff
@@ -0,0 +1,107 @@
+module InPerson
+  # This job checks a queue regularly to determine whether USPS has notitied us
+  # about whether an in-person enrollment is ready to have its status checked. If
+  # the enrollment is ready, then this job updates a flag on the enrollment so that it
+  # will be checked earlier than other enrollments.
+  class EnrollmentsReadyForStatusCheckJob < ApplicationJob
+    queue_as :low
+
+    def perform(_now)
+      return true unless IdentityConfig.store.in_person_proofing_enabled &&
+                         IdentityConfig.store.in_person_enrollments_ready_job_enabled
+
+      begin
+        analytics.idv_in_person_proofing_enrollments_ready_for_status_check_job_started
+
+        analytics_stats = {
+          fetched_items: 0,
+          processed_items: 0,
+          deleted_items: 0,
+          valid_items: 0,
+          invalid_items: 0,
+        }
+
+        # Continually request messages until no messages are received
+        while (messages = poll).any?
+          process_batch(messages, analytics_stats)
+        end
+        return true
+      ensure
+        analytics.idv_in_person_proofing_enrollments_ready_for_status_check_job_completed(
+          **analytics_stats,
+          incomplete_items:
+            analytics_stats[:fetched_items] - analytics_stats[:processed_items],
+          deletion_failed_items:
+            analytics_stats[:processed_items] - analytics_stats[:deleted_items],
+        )
+      end
+    end
+
+    private
+
+    delegate :poll, to: :sqs_batch_wrapper
+    delegate :process_batch, to: :batch_processor
+    delegate :analytics, to: :analytics_factory
+
+    def batch_processor
+      @batch_processor ||= begin
+        InPerson::EnrollmentsReadyForStatusCheck::BatchProcessor.new(
+          error_reporter: InPerson::EnrollmentsReadyForStatusCheck::ErrorReporter.new(
+            InPerson::EnrollmentsReadyForStatusCheck::BatchProcessor.name,
+            analytics,
+          ),
```

Comment:
I like using dependency injection, but I think that it seems redundant to pass in the `class_name` explicitly like this, could the `BatchProcessor` class have a default implementation that passes its own `self.class.name` in to the error reporter? Since `analytics` here is basically just a default/empty instance with no user, I feel like we don't need to worry about passing in the same one necessarily

Ditto for below with `EnrollmentPipeline`

Reply:
Default implementations for DI have been a colossal headache for me in the past, so I really prefer to avoid default implementations.